presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -1187,6 +1189,18 @@ public SystemSessionProperties(
                         KEY_BASED_SAMPLING_FUNCTION,
                         "Sampling function for key based sampling",
                         "key_sampling_percent",
+                        false),
+                integerProperty(
+                        HASH_BASED_DISTINCT_LIMIT_THRESHOLD,
+                        "threshold for doing hash based distinct",
+                        // 10k is a decent-sized limit that guarantees almost no collisions if the # distinct < 10k according to:
```

Comment:
Should we add a check that the threshold is < 10K? I'm concerned about users inadvertently configuring the system in a way that may produce approximate results. Investigate such issues will be extremely hard. I think having a top-level config to opt-in into approximate results would be helpful. It may be helpful to also issue a warning whenever a query is using a feature / optimization that may produce approximate result.

Reply:
This is not a config but rather query level session param. So as long as it doesn't OOM, I see no issue (again it says the chance of collision starts at 2bn - 4bn distinct values.

Reply:
I'm a bit confused here by the statement around threshold for collision - should we assume this is == 0 or ~= 0 at sufficiently low cardinalities (like the 10K proposed above)?

Reply:
> I'm a bit confused here by the statement around threshold for collision - should we assume this is == 0 or ~= 0 at sufficiently low cardinalities (like the 10K proposed above)?

Yes that's the idea. 10k is what we are starting with but if other deployments want to change it, they can do it.